### PR TITLE
Support mortar meshes without corresponding mortar constraints

### LINKED ISSUE #20214: Allow having mortar meshes without corresponding mortar constraints
## Reason
With the creation of mortar aux kernels, it is theoretically possible, for example, that a user wants to compute an aux quantity on a mortar interface in the reference configuration whereas they want the mortar constraint to execute in the displaced configuration. Currently our logic in the constraint warehouse and `NonlinearSystemBase` doesn't support this use.

## Design
Create `hasActiveMortarConstraints` method in the `ConstraintWarehouse` and then some small other changes.

## Impact
Allow seamless combinations of reference and displaced mesh mortar computations.

### PR DESCRIPTION
**Pull request by lindsayad**
Closes #20214

### CODE REVIEW

File: framework/src/systems/NonlinearSystemBase.C [hunk 1]
```diff
@@ -3311,33 +3317,11 @@ NonlinearSystemBase::setPreviousNewtonSolution(const NumericVector<Number> & sol
...
-
-    mooseAssert(it != _displaced_mortar_functors.end(),
-                "No ComputeMortarFunctor exists for the specified primary-secondary boundary "
-                "pair, primary "
-                    << mortar_interface.first.first << " and secondary "
-                    << mortar_interface.first.second);
-    it->second();
-  }
+  for (auto & map_pr : _displaced_mortar_functors)
+    map_pr.second();
```
Comment (recuero) [inline framework/src/systems/NonlinearSystemBase.C:3324]: Now, we don't add the check that each functor must have a particular primary-secondary pair. I can't see if the added code is taking care of possible error.

Comment (lindsayad) [reply framework/src/systems/NonlinearSystemBase.C:3324]: What possible error? We should only be looping over primary-secondary pairs that were requested by the user through creation of mortar aux kernels and mortar constraints. So there should be no possibility of an error here

Comment (recuero) [reply framework/src/systems/NonlinearSystemBase.C:3324]: Ok. Was referring to the removal of the assertion:

``` mooseAssert(it != _displaced_mortar_functors.end(),
                "No ComputeMortarFunctor exists for the specified primary-secondary boundary "
                "pair, primary "
                    << mortar_interface.first.first << " and secondary "
                    << mortar_interface.first.second);

File: framework/src/systems/NonlinearSystemBase.C [hunk 2]
```diff
@@ -294,7 +294,10 @@ NonlinearSystemBase::initialSetup()
         _fe_problem.getMortarInterfaces(/*displaced=*/false);
     for (const auto & mortar_interface : undisplaced_mortar_interfaces)
     {
-      auto primary_secondary_boundary_pair = mortar_interface.first;
+      const auto primary_secondary_boundary_pair = mortar_interface.first;
+      if (!_constraints.hasActiveMortarConstraints(primary_secondary_boundary_pair, false))
+        continue;
```
Comment (lindsayad) [inline framework/src/systems/NonlinearSystemBase.C:299]: @recuero this is the logic change that removes the failed assertions

Comment (recuero) [reply framework/src/systems/NonlinearSystemBase.C:299]: Got it. Whether or not we have mortar aux kernels doesn't really matter for this code. Here, only constraints are dealt with.

**Review by recuero — APPROVED** [review]

**Pull request merged**